From 6978a8aa9f80726db8594933e2b75340823a3cce Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Thu, 28 Mar 2013 09:54:04 +0800
Subject: [PATCH] powerpc/mpic_timer: convert sysdev_class to a regular subsystem

Kay Sievers introduced the commit ca22e56d (driver-core: implement
'sysdev' functionality for regular devices and buses) and converted
all the sysdev classes and sysdev devices to regular buses and
devices. The drivers/base/sys.c and include/linux/sysdev.h was
removed from the kernel finally. Make the same conversion for the
mpic timer driver according to this change.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/sysdev/fsl_mpic_timer_wakeup.c |   25 +++++++++++++------------
 1 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/arch/powerpc/sysdev/fsl_mpic_timer_wakeup.c b/arch/powerpc/sysdev/fsl_mpic_timer_wakeup.c
index 53d0732..0625c79 100644
--- a/arch/powerpc/sysdev/fsl_mpic_timer_wakeup.c
+++ b/arch/powerpc/sysdev/fsl_mpic_timer_wakeup.c
@@ -17,7 +17,7 @@
 #include <linux/interrupt.h>
 #include <linux/sysfs.h>
 #include <linux/kobject.h>
-#include <linux/sysdev.h>
+#include <linux/device.h>
 
 #include <asm/mpic_timer.h>
 
@@ -46,8 +46,8 @@ static irqreturn_t fsl_mpic_timer_irq(int irq, void *dev_id)
 	return IRQ_HANDLED;
 }
 
-static ssize_t fsl_timer_wakeup_show(struct sysdev_class *kobj,
-				struct sysdev_class_attribute *attr,
+static ssize_t fsl_timer_wakeup_show(struct device *dev,
+				struct device_attribute *attr,
 				char *buf)
 {
 	struct timeval interval;
@@ -68,8 +68,8 @@ static ssize_t fsl_timer_wakeup_show(struct sysdev_class *kobj,
 	return status;
 }
 
-static ssize_t fsl_timer_wakeup_store(struct sysdev_class *kobj,
-				struct sysdev_class_attribute *attr,
+static ssize_t fsl_timer_wakeup_store(struct device *dev,
+				struct device_attribute *attr,
 				const char *buf, size_t count)
 {
 	struct timeval interval;
@@ -97,11 +97,12 @@ static ssize_t fsl_timer_wakeup_store(struct sysdev_class *kobj,
 	return count;
 }
 
-static SYSDEV_CLASS_ATTR(timer_wakeup, 0644,
+static DEVICE_ATTR(timer_wakeup, S_IWUSR | S_IRGRP | S_IROTH,
 			fsl_timer_wakeup_show, fsl_timer_wakeup_store);
 
-static struct sysdev_class mpic_sysdev_class = {
+static struct bus_type mpic_subsys = {
 	.name = "mpic",
+	.dev_name = "mpic",
 };
 
 static int __init fsl_wakeup_sys_init(void)
@@ -114,13 +115,13 @@ static int __init fsl_wakeup_sys_init(void)
 
 	INIT_WORK(&fsl_wakeup->free_work, fsl_free_resource);
 
-	ret = sysdev_class_register(&mpic_sysdev_class);
+	ret = subsys_system_register(&mpic_subsys, NULL);
 	if (ret)
 		goto err;
 
-	ret = sysdev_class_create_file(&mpic_sysdev_class, &attr_timer_wakeup);
+	ret = device_create_file(mpic_subsys.dev_root, &dev_attr_timer_wakeup);
 	if (ret) {
-		sysdev_class_unregister(&mpic_sysdev_class);
+		bus_unregister(&mpic_subsys);
 		goto err;
 	}
 
@@ -133,9 +134,9 @@ err:
 
 static void __exit fsl_wakeup_sys_exit(void)
 {
-	sysdev_class_remove_file(&mpic_sysdev_class, &attr_timer_wakeup);
+	device_remove_file(mpic_subsys.dev_root, &dev_attr_timer_wakeup);
 
-	sysdev_class_unregister(&mpic_sysdev_class);
+	bus_unregister(&mpic_subsys);
 
 	kfree(fsl_wakeup);
 }
-- 
1.7.0

